lib/assets/data_table/main.js

```diff
@@ -1,163 +1,310 @@
+    const [showSearch, setShowSearch] = useState(false);
+    const [columns, setColumns] = useState(columnsInitData);
+    const [colSizes, setColSizes] = useState(columnsInitSize);
 
-      refetch() {
-        this.loading = true;
-        ctx.pushEvent("refetch");
-      },
+    const infiniteScroll = content.limit === totalRows;
+    const height = totalRows >= 10 && infiniteScroll ? 484 : null;
 
-      orderBy(newKey) {
-        this.loading = true;
-        const [key, order] = reorder(this.data.content.order_by, this.data.content.order, newKey)
-        ctx.pushEvent("order_by", { key, order });
-      },
+    const rows =
+      content.page === content.max_page && !infiniteScroll
+        ? totalRows % content.limit
+        : content.limit;
+
+    const getData = useCallback(
+      ([col, row]) => {
+        const rawData = content.rows[row].fields[col];
+        const kind = cellKind[content.columns[col].type] || GridCellKind.Text;
 
-      orderDescription() {
-        const key = this.data.content.order_by;
-        const { label } = this.data.content.columns.find((column) => column.key === key);
-        const order = this.data.content.order === "asc" ? "ascending" : "descending";
-        return `ordered by ${label}, ${order}`;
+        const cellData = rawData.startsWith('"')
+          ? rawData.slice(1, -1)
+          : rawData;
+
+        return {
+          kind: kind,
+          data: cellData,
+          displayData: cellData,
+          allowOverlay: true,
+          allowWrapping: true,
+          readonly: true,
+        };
       },
-    },
-
-    components: {
-      RemixIcon: {
-        props: ["icon"],
-        template: `
-          <i :class="'ri ri-' + icon" aria-hidden="true"></i>
-        `
-      }
-    }
-  }).mount(ctx.root);
+      [content]
+    );
 
-  ctx.handleEvent("update_content", (content) => {
-    app.data.content = content;
-    app.loading = false;
-  });
+    const toggleSearch = () => {
+      setShowSearch(!showSearch);
+    };
+
+    const orderBy = (colIndex) => {
+      const newKey = columns[colIndex].id;
+      const [key, order] = reorder(content.order_by, content.order, newKey);
+      ctx.pushEvent("order_by", { key, order });
+    };
+
+    const onColumnResize = useCallback((column, newSize) => {
+      setColSizes((prevColSizes) => {
+        return {
+          ...prevColSizes,
+          [column.title]: newSize,
+        };
+      });
+    }, []);
+
+    useEffect(() => {
+      ctx.handleEvent("update_content", (content) => {
+        setContent(content);
+      });
+    }, []);
+
+    useEffect(() => {
+      const icon = content.order === "asc" ? "arrowUp" : "arrowDown";
+      const newColumns = columns.map((header) => {
+        if (header.id === content.order_by) {
+          return { ...header, overlayIcon: icon };
+        } else {
+          return { ...header, overlayIcon: null };
+        }
+      });
+      setColumns(newColumns);
+    }, [content]);
```

Comment:
We can be more specific to avoid reevaluations, especially that content is an object

```suggestion
    }, [content.order, content.order_by]);
```